src/bci_build/package/appcontainers.py

```diff
@@ -808,3 +809,82 @@ def _get_nginx_kwargs(os_version: OsVersion):
     )
     for tomcat_major, os_version in product(_TOMCAT_VERSIONS, ALL_BASE_OS_VERSIONS)
 ]
+
+SPACK_CONTAINERS = [
+    ApplicationStackContainer(
+        name="spack",
+        package_name="spack-image",
+        pretty_name="Spack package manager for supercomputers - build container",
+        os_version=os_version,
+        is_latest=os_version in CAN_BE_LATEST_OS_VERSION,
+        other_tags=["next"]
+        if not (
+            (os_version in CAN_BE_LATEST_OS_VERSION)
+            or (os_version in RELEASED_OS_VERSIONS)
+        )
+        else None,
+        version="%%spack_version%%",
+        version_in_uid=False,
+        replacements_via_service=[
+            Replacement(
+                regex_in_build_description="%%spack_version%%",
+                package_name="spack",
+                parse_version="patch",
+            ),
+            Replacement(
+                regex_in_build_description="%%spack_version%%",
+                file_name="README.md",
+                package_name="spack",
+                parse_version="patch",
+            ),
+        ],
+        package_list=[
+            "spack",
+            "autoconf",
+            "bison",
+            "ccache",
+            "cmake-full",
+            "flex",
+            "libtool",
+            "libzip-devel",
+            "libcurl-devel",
+            "libopenssl-devel",
+            "m4",
+            "makeinfo",
+            "ncurses-devel",
+            "patchelf",
+            "python3-base",
+            "python3-boto3",
+            "tack",
+            "xz-devel",
+        ],
```

Comment:
You could try to disable `--no-recommends`, if spack recommends most of these packages, then you might be able to cull the list a bit
```suggestion
        package_list=[
            "spack",
            "autoconf",
            "bison",
            "ccache",
            "cmake-full",
            "flex",
            "libtool",
            "libzip-devel",
            "libcurl-devel",
            "libopenssl-devel",
            "m4",
            "makeinfo",
            "ncurses-devel",
            "patchelf",
            "python3-base",
            "python3-boto3",
            "tack",
            "xz-devel",
        ],
        no_recommends = False,
```

Reply:
I've tried this - and removed all but `spack` from the list. I'm still waiting for IBS to finally publish this - for over one hour now. I've had enough I'm not wasting another evening on waiting for the build service.

Reply:
Ok, I've reduced the list. Relying on OBS to handle recommends correctly did not work. 
I had to add recommended packages explicitly. Also, I had to include `tack` explicitly - although `ncurses-devel` requires it.
Unfortunately, I also needed to add `lsb-release`, `zstd` and `patchelf` for now. Once we release the next update to Spack this should be fixed, with updating Spack I want to wait for 0.21.3 to be released. This should happen in early.